modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsaMPCv2.ts

```diff
@@ -368,6 +369,50 @@ export class EcdsaMPCv2Utils extends BaseEcdsaUtils {
     return { ...(await keychains.add(recipientKeychainParams)), reducedEncryptedPrv: reducedEncryptedPrv };
   }
 
+  async getMpcV2RetrofitDataFromMpcV1Keys(params: {
+    mpcv1UserKeyShare: string;
+    mpcv1BackupKeyShare: string;
+    bitgoBigSi: string;
+  }): Promise<{ mpcv2UserKeyShare: DklsTypes.RetrofitData; mpcv2BakcupKeyShare: DklsTypes.RetrofitData }> {
+    const userSigningMaterial: ECDSAMethodTypes.SigningMaterial = JSON.parse(params.mpcv1UserKeyShare);
+    const backupSigningMaterial: ECDSAMethodTypes.SigningMaterial = JSON.parse(params.mpcv1BackupKeyShare);
+    const mpc = new Ecdsa();
+    assert(userSigningMaterial.backupNShare, 'User MPCv1 key material should have backup NShare.');
+    assert(backupSigningMaterial.userNShare, 'Backup MPCv1 key material should have user NShare.');
+    const userCombined = mpc.keyCombine(userSigningMaterial.pShare, [
+      userSigningMaterial.backupNShare,
+      userSigningMaterial.bitgoNShare,
+    ]);
+    const backupCombined = mpc.keyCombine(backupSigningMaterial.pShare, [
+      backupSigningMaterial.userNShare,
+      backupSigningMaterial.bitgoNShare,
+    ]);
+    const secp256k1 = new Secp256k1Curve();
+    const userBigSi = bigIntToBufferBE(secp256k1.basePointMult(BigInt('0x' + userCombined.xShare.x))).toString('hex');
+    const backupBigSi = bigIntToBufferBE(secp256k1.basePointMult(BigInt('0x' + backupCombined.xShare.x))).toString(
+      'hex'
+    );
+    const xiList = [
+      Array.from(bigIntToBufferBE(BigInt(1), 32)),
+      Array.from(bigIntToBufferBE(BigInt(2), 32)),
+      Array.from(bigIntToBufferBE(BigInt(3), 32)),
+    ];
+    const bigSiListUser = [backupBigSi, params.bitgoBigSi];
+    const bigSiListBackup = [userBigSi, params.bitgoBigSi];
+    return {
+      mpcv2BakcupKeyShare: {
+        bigSiList: bigSiListBackup,
+        xShare: backupCombined.xShare,
+        xiList: xiList,
+      },
+      mpcv2UserKeyShare: {
+        bigSiList: bigSiListUser,
+        xShare: userCombined.xShare,
+        xiList: xiList,
+      },
```

Comment:
just confirming, this is the payload im goin to need add as the retrofitData for the dkls dkg methods?

Reply:
yes